### DISCUSSION TITLE
"Sendmail":"Memory error"   esp8266

### DISCUSSION THREAD

**Discussion by DC-555** [arendst/Tasmota#19364] [Q&A]
Hello, I'm trying to send emails from my 8266, but I get this error message:
Observation, I have a similar configuration with the same user and password in esp32 and it works fine.

RESULT = {"Sendmail":"Memory error"}

Here is my email configuration:

sendmail [smtp.gmail.com:465:mymail@gmail.com:passwordAPP:<mymail@gmail.com>:<sendmail@gmail.com:alarm] test

user_config_override.h :
#define USE_SENDMAIL
#define USE_TLS

**Comment by barbudor** [comment]
May be TLS and SEND_MAIL are too memory consuming features to live together ?
You may want to remove other unecessary features.
Look at how tasmota-lite is done and start to #undef features that you don't need

**Reply by sfromis** [reply]
Or switch to an ESP32.

**Comment by DC-555** [comment]
@barbudor 
Follows my USER_CONFIG_OVERRIDE_H, but I already disabled a lot.
`
#ifndef _USER_CONFIG_OVERRIDE_H_
#define _USER_CONFIG_OVERRIDE_H_

//*****************************************************************************************************
// -- Configurações do WIFI  ---------------
#undef  STA_SSID1
#define STA_SSID1              "#####"                // [Ssid1] Wi-Fi SSID
#undef STA_PASS1
#define STA_PASS1              "#####"                // [Password1] Wi-Fi password
#undef STA_SSID2
#define STA_SSID2              "####"                // [Ssid2] Optional alternate AP Wi-Fi SSID
#undef STA_PASS2
#define STA_PASS2              "####"                // [Password2] Optional alternate AP Wi-Fi password
// -----------------------------------------
// -- Configurações do MQTT  ---------------
#undef MQTT_KEEPALIVE
#define MQTT_KEEPALIVE         10                // [MqttKeepAlive] Number of seconds between KeepAlive messages
#undef MQTT_HOST
#define MQTT_HOST              "####"                // [MqttHost]
#undef MQTT_PORT
#define MQTT_PORT              8883              // [MqttPort] MQTT port (10123 on CloudMQTT)
#undef MQTT_USER
#define MQTT_USER              "####"       // [MqttUser] MQTT user
#undef MQTT_PASS
#define MQTT_PASS              "####"       // [MqttPassword] MQTT password
// -- Configurações adicionais de MQTT ----
#undef MQTT_GRPTOPIC
#define MQTT_GRPTOPIC          "####"        // [GroupTopic] MQTT Group topic
#undef TELE_PERIOD
#define TELE_PERIOD            60               // [TelePeriod] Telemetry (0 = disable, 10 - 3600 seconds)
//------------------------------------------
// -- Configurações gerais  ----------------
#ifdef GUI_SHOW_HOSTNAME
#undef GUI_SHOW_HOSTNAME
#endif
#define GUI_SHOW_HOSTNAME      true
//----------
#define MY_LANGUAGE            pt_BR           // Portuguese in Brazil
#undef WEB_PASSWORD
#define WEB_PASSWORD           ""                // [WebPassword] Web server Admin mode Password for WEB_USERNAME (empty string = Disable)
#undef FRIENDLY_NAME
#define FRIENDLY_NAME          "####"         // [FriendlyName] Friendlyname up to 32 characters used by webpages and Alexa
//------------------------------------------
//-- Configuracao visuais ---------------------------
#undef COLOR_BUTTON
#define COLOR_BUTTON                "#038c6b"    // [WebColor11] cor do botoo verde axon
//---------------------------------------------------
//-- Configuração da Regiao -------------------------
//Removendo parte antiga
#undef TIME_DST_HEMISPHERE
#undef TIME_DST_WEEK
#undef TIME_DST_DAY
#undef TIME_DST_MONTH
#undef TIME_DST_HOUR
#undef TIME_DST_OFFSET
// -- Time - Start Standard Time and timezone offset from UTC in minutes
#undef TIME_STD_HEMISPHERE
#undef TIME_STD_WEEK
#undef TIME_STD_DAY
#undef TIME_STD_MONTH
#undef TIME_STD_HOUR
#undef TIME_STD_OFFSET
// -- Location ------------------------------------
#undef LATITUDE
#undef LONGITUDE
#undef APP_TIMEZONE

// -- adicionando nova configuracao ---------------
#define TIME_DST_HEMISPHERE    1             // [TimeDst] Hemisphere (0 or North, 1 or South)
#define TIME_DST_WEEK          0              // Week of month (0 or Last, 1 or First, 2 or Second, 3 or Third, 4 or Fourth)
#define TIME_DST_DAY           1               // Day of week (1 or Sun, 2 or Mon, 3 or Tue, 4 or Wed, 5 or Thu, 6 or Fri, 7 or Sat)
#define TIME_DST_MONTH         10               // Month (1 or Jan, 2 or Feb, 3 or Mar, 4 or Apr, 5 or May, 6 or Jun, 7 or Jul, 8 or Aug, 9 or Sep, 10 or Oct, 11 or Nov, 12 or Dec)
#define TIME_DST_HOUR          2                 // Hour (0 to 23)
#define TIME_DST_OFFSET        -120              // Offset from UTC in minutes (-780 to +780)
// -- Time - Start Standard Time and timezone offset from UTC in minutes
#define TIME_STD_HEMISPHERE    1             // [TimeStd] Hemisphere (0 or North, 1 or South)
#define TIME_STD_WEEK          0              // Week of month (0 or Last, 1 or First, 2 or Second, 3 or Third, 4 or Fourth)
#define TIME_STD_DAY           1               // Day of week (1 or Sun, 2 or Mon, 3 or Tue, 4 or Wed, 5 or Thu, 6 or Fri, 7 or Sat)
#define TIME_STD_MONTH         2               // Month (1 or Jan, 2 or Feb, 3 or Mar, 4 or Apr, 5 or May, 6 or Jun, 7 or Jul, 8 or Aug, 9 or Sep, 10 or Oct, 11 or Nov, 12 or Dec)
#define TIME_STD_HOUR          3                 // Hour (0 to 23)
#define TIME_STD_OFFSET        -180              // Offset from UTC in minutes (-780 to +780)
// -- Location ------------------------------------
#define LATITUDE               -23.550520        // [Latitude] Your location to be used with sunrise and sunset
#define LONGITUDE              -46.633308        // [Longitude] Your location to be used with sunrise and sunset
// -- Application ---------------------------------
#define APP_TIMEZONE           99                // [Timezone] Use TIME_DST and TIME_STD for DST and Standard Time
#undef WIFI_DNS
#define WIFI_DNS              "time.google.com"
//--------------------------------------------------
// -- Configuração dos Switch ----------------------
#undef SWITCH_DEBOUNCE_TIME
#define SWITCH_DEBOUNCE_TIME   15                // [SwitchDebounce] Number of mSeconds switch press debounce time
#undef SWITCH_MODE
#define SWITCH_MODE            FOLLOW_INV            // [SwitchMode] TOGGLE, FOLLOW, FOLLOW_INV, PUSHBUTTON, PUSHBUTTON_INV, PUSHBUTTONHOLD, PUSHBUTTONHOLD_INV, PUSHBUTTON_TOGGLE, TOGGLEMULTI, FOLLOWMULTI, FOLLOWMULTI_INV (the wall switch state)
#undef MQTT_SWITCHES
#define MQTT_SWITCHES          true             // [SetOption114] Detach switches from relays and send MQTT messages instead
//--------------------------------------------------
// -- Configuração de Script -----------------------
#ifdef USE_RULES
#undef USE_RULES                                // Add support for rules (+8k code)
#endif
#ifdef USE_SCRIPT
#undef USE_SCRIPT
#endif
#define USE_SCRIPT
#define USE_SCRIPT_FATFS 4                     // Script: Add FAT FileSystem Support
#define SUPPORT_MQTT_EVENT                     // Support trigger event with MQTT subscriptions (+3k5 code)

#define USE_WEBGETCONFIG
#define SCRIPT_MAXSSIZE 156
#define USE_SCRIPT_FATFS_EXT
//#define USE_SCRIPT_SERIAL

//--------------------------------------------------
// -- Configuração de E-mail -----------------------
#define USE_SENDMAIL
#define USE_TLS 
 #define EMAIL_PASSWORD "llzaypwarmlfhkkm"
//--------------------------------------------------
//-- Configuração do RFID RDM6300 ------------------
#ifdef USE_RDM6300 
#define USE_RDM6300			// Add support for RDM6300 125kHz RFID Reader (+0k8)
#endif
//--------------------------------------------------
//-- Configuração do ger. de aqruivos --------------
#ifdef USE_UFILESYS
  #undef USE_UFILESYS
#endif
#define USE_UFILESYS

#ifdef GUI_TRASH_FILE
  #undef GUI_TRASH_FILE
#endif
#define GUI_TRASH_FILE

#ifdef GUI_EDIT_FILE
  #undef GUI_EDIT_FILE
#endif
#define GUI_EDIT_FILE
//--------------------------------------------------
// -- Configuração do Mod bus ----------------------
#ifndef USE_SERIAL_BRIDGE
#define USE_SERIAL_BRIDGE                        // Add support for software Serial Bridge (+2k code)
#endif
#define USE_MODBUS_BRIDGE
#define SERIAL_BRIDGE_BUFFER_SIZE 256
//--------------------------------------------------
// -- Configuração de template e tópico ------------
// -- Project -------------------------------------
#ifdef PROJECT
#undef PROJECT
#endif
#define PROJECT                "connect"         // PROJECT is used as the default topic delimiter
// If not selected the default will be SONOFF_BASIC
//#define MODULE                 SONOFF_BASIC      // [Module] Select default module from tasmota_template.h
#ifdef ESP8266
#undef FALLBACK_MODULE
#define FALLBACK_MODULE        USER_MODULE      // [Module2] Select default module on fast reboot where USER_MODULE is user template
#define USER_TEMPLATE "{\"NAME\":\"####2\",\"GPIO\":[1,8768,1,8800,225,224,1,1,194,195,193,576,192,1],\"FLAG\":0,\"BASE\":18,\"CMND\":\"module 0\"}"  // [Template] Set JSON template
#endif  // ESP8266
#ifdef MQTT_GRPTOPIC
#undef MQTT_GRPTOPIC
#endif
#define MQTT_GRPTOPIC          "####"        // [GroupTopic] MQTT Group topic
//------------------------------------------------------
// -- Configuração de desativar itens que não usa ------
#ifdef DOMOTICZ_UPDATE_TIMER
#undef DOMOTICZ_UPDATE_TIMER
#endif

#ifdef HOME_ASSISTANT_DISCOVERY_ENABLE
#undef HOME_ASSISTANT_DISCOVERY_ENABLE
#endif

#ifdef USE_DOMOTICZ
#undef USE_DOMOTICZ
#endif

#ifdef USE_TELEGRAM_FINGERPRINT
#undef USE_TELEGRAM_FINGERPRINT
#endif

#ifdef USE_KNX_WEB_MENU
#undef USE_KNX_WEB_MENU
#endif

#ifdef USE_TIMERS
#undef USE_TIMERS
#endif

#ifdef ROTARY_V1
#undef ROTARY_V1
#endif

#ifdef USE_SONOFF_RF
#undef USE_SONOFF_RF
#endif

#ifdef USE_SONOFF_SC
#undef USE_SONOFF_SC
#endif

#ifdef USE_TUYA_MCU
#undef USE_TUYA_MCU
#endif

#ifdef USE_ARMTRONIX_DIMMERS
#undef USE_ARMTRONIX_DIMMERS
#endif

#ifdef USE_PS_16_DZ
#undef USE_PS_16_DZ
#endif

#ifdef USE_SONOFF_IFAN
#undef USE_SONOFF_IFAN
#endif

#ifdef USE_BUZZER
#undef USE_BUZZER
#endif

#ifdef USE_ARILUX_RF
#undef USE_ARILUX_RF
#endif

#ifdef USE_SHUTTER
#undef USE_SHUTTER
#endif

#ifdef USE_DEEPSLEEP
#undef USE_DEEPSLEEP
#endif

#ifdef USE_EXS_DIMMER
#undef USE_EXS_DIMMER
#endif

#ifdef USE_DEVICE_GROUPS
#undef USE_DEVICE_GROUPS
#endif

#ifdef USE_PWM_DIMMER
#undef USE_PWM_DIMMER
#endif

#ifdef USE_SONOFF_D1
#undef USE_SONOFF_D1
#endif

#ifdef USE_SHELLY_DIMMER
#undef USE_SHELLY_DIMMER
#endif

#ifdef SHELLY_CMDS
#undef SHELLY_CMDS
#endif

#ifdef SHELLY_FW_UPGRADE
#undef SHELLY_FW_UPGRADE
#endif

#ifdef USE_LIGHT
#undef USE_LIGHT
#endif

#ifdef USE_WS2812
#undef USE_WS2812
#endif

#ifdef USE_WS2812_RMT
#undef USE_WS2812_RMT
#endif

#ifdef USE_WS2812_HARDWARE
#undef USE_WS2812_HARDWARE
#endif

#ifdef USE_WS2812_CTYPE
#undef USE_WS2812_CTYPE
#endif

#ifdef USE_MY92X1
#undef USE_MY92X1
#endif

#ifdef USE_SM16716
#undef USE_SM16716
#endif

#ifdef USE_SM2135
#undef USE_SM2135
#endif

#ifdef USE_SM2335
#undef USE_SM2335
#endif

#ifdef USE_BP1658CJ
#undef USE_BP1658CJ
#endif

#ifdef USE_BP5758D
#undef USE_BP5758D
#endif

#ifdef USE_SONOFF_L1
#undef USE_SONOFF_L1
#endif

#ifdef USE_ELECTRIQ_MOODL
#undef USE_ELECTRIQ_MOODL
#endif

#ifdef USE_LIGHT_PALETTE
#undef USE_LIGHT_PALETTE
#endif

#ifdef USE_LIGHT_VIRTUAL_CT
#undef USE_LIGHT_VIRTUAL_CT
#endif

#ifdef USE_DGR_LIGHT_SEQUENCE
#undef USE_DGR_LIGHT_SEQUENCE
#endif

#ifdef USE_LIGHT_ARTNET_MCAST
#undef USE_LIGHT_ARTNET_MCAST
#endif

#ifdef USE_COUNTER
#undef USE_COUNTER
#endif

#ifdef USE_DS18x20
#undef USE_DS18x20
#endif

#ifdef USE_I2C
#undef USE_I2C
#endif

#ifdef MAX_TOUCH_BUTTONS
#undef MAX_TOUCH_BUTTONS
#endif

#ifdef USE_MIBLE
#undef USE_MIBLE
#endif

#ifdef USE_DISPLAY_ILI9341
#undef USE_DISPLAY_ILI9341
#endif

#ifdef CO2_LOW
#undef CO2_LOW
#endif

#ifdef CO2_HIGH
#undef CO2_HIGH
#endif

#ifdef STARTING_OFFSET
#undef STARTING_OFFSET
#endif

#ifdef MP3_VOLUME
#undef MP3_VOLUME
#endif

#ifdef USE_ENERGY_SENSOR
#undef USE_ENERGY_SENSOR
#endif

#ifdef USE_ENERGY_MARGIN_DETECTION
#undef USE_ENERGY_MARGIN_DETECTION
#endif

#ifdef USE_ENERGY_POWER_LIMIT
#undef USE_ENERGY_POWER_LIMIT
#endif

#ifdef USE_ENERGY_DUMMY
#undef USE_ENERGY_DUMMY
#endif

#ifdef USE_HLW8012
#undef USE_HLW8012
#endif

#ifdef USE_CSE7766
#undef USE_CSE7766
#endif

#ifdef USE_PZEM004T
#undef USE_PZEM004T
#endif

#ifdef USE_PZEM_AC
#undef USE_PZEM_AC
#endif

#ifdef USE_PZEM_DC
#undef USE_PZEM_DC
#endif

#ifdef USE_MCP39F501
#undef USE_MCP39F501
#endif

#ifdef SDM72_SPEED
#undef SDM72_SPEED
#endif

#ifdef SDM120_SPEED
#undef SDM120_SPEED
#endif

#ifdef SDM230_SPEED
#undef SDM230_SPEED
#endif

#ifdef SDM630_SPEED
#undef SDM630_SPEED
#endif

#ifdef DDS2382_SPEED
#undef DDS2382_SPEED
#endif

#ifdef DDSU666_SPEED
#undef DDSU666_SPEED
#endif

#ifdef SOLAXX1_SPEED
#undef SOLAXX1_SPEED
#endif

#ifdef LE01MR_SPEED
#undef LE01MR_SPEED
#endif

#ifdef LE01MR_ADDR
#undef LE01MR_ADDR
#endif

#ifdef USE_BL09XX
#undef USE_BL09XX
#endif

#ifdef IEM3000_SPEED
#undef IEM3000_SPEED
#endif

#ifdef IEM3000_ADDR
#undef IEM3000_ADDR
#endif

#ifdef USE_DHT
#undef USE_DHT
#endif

#ifdef USE_IR_REMOTE
#undef USE_IR_REMOTE
#endif

#ifdef USE_IR_RECEIVE
#undef USE_IR_RECEIVE
#endif

#ifdef SDC_HIDE_INVISIBLES
#undef SDC_HIDE_INVISIBLES
#endif

#ifdef USE_ZIGBEE_ZNP
#undef USE_ZIGBEE_ZNP
#endif

#ifdef USE_ZIGBEE_CHANNEL
#undef USE_ZIGBEE_CHANNEL
#endif

#ifdef USE_ZIGBEE_TXRADIO_DBM
#undef USE_ZIGBEE_TXRADIO_DBM
#endif

#ifdef USE_ZIGBEE_COALESCE_ATTR_TIMER
#undef USE_ZIGBEE_COALESCE_ATTR_TIMER
#endif

#ifdef USE_ZIGBEE_DEBOUNCE_COMMANDS
#undef USE_ZIGBEE_DEBOUNCE_COMMANDS
#endif

#ifdef USE_ZIGBEE_MODELID
#undef USE_ZIGBEE_MODELID
#endif

#ifdef USE_ZIGBEE_MANUFACTURER
#undef USE_ZIGBEE_MANUFACTURER
#endif

#ifdef USE_ZIGBEE_BATT_REPROBE
#undef USE_ZIGBEE_BATT_REPROBE
#endif

#ifdef USE_ZIGBEE_BATT_REPROBE_PAUSE
#undef USE_ZIGBEE_BATT_REPROBE_PAUSE
#endif

#ifdef USE_ZBBRIDGE_TLS
#undef USE_ZBBRIDGE_TLS
#endif

#ifdef USE_ZIGBEE_ZBBRIDGE_EEPROM
#undef USE_ZIGBEE_ZBBRIDGE_EEPROM
#endif

#ifdef USE_ZIGBEE_AUTOBIND_MAX_ENDPOINTS
#undef USE_ZIGBEE_AUTOBIND_MAX_ENDPOINTS
#endif

#ifdef USE_ZIGBEE_AUTOBIND_MAX_ENDPOINTS
#undef USE_ZIGBEE_AUTOBIND_MAX_ENDPOINTS
#endif

#ifdef USE_ZIGBEE_AUTOBIND_MAX_CLUSTER
#undef USE_ZIGBEE_AUTOBIND_MAX_CLUSTER
#endif

// -----------------------------------------------------

#endif  // Fim do Override.h
`

**Reply by sfromis** [reply]
Running the tool [decode-status](https://github.com/tasmota/decode-status), you can instead see what you still have enabled.... Much more readable.

Still, using TLS is a major memory hog, not leaving much room for other memory-heavy features. ESP8266 is too limited for combining complex features.

**Reply by DC-555** [reply]
I'll do the decode-status, but I still have a lot of free memory:

11:02:32.912 MQT: stat/connect_DF067A/STATUS4 = {"StatusMEM":{"ProgramSize":671,"Free":1376,"Heap":11,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"16405E","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000416","0080558A","042E0001","00000003","00000000","C000E881","00000084","00201000","54000000","00000080"],"Drivers":"1,2,4,6,8,10,12,18,19,22,24,26,29,30,35,37,38,45,50,62,63,68","Sensors":"1,2"}}

**Reply by DC-555** [reply]
decode:

Options
    0 (ON ) (Settings) Save power state (1) and use after restart
    1 (OFF) (Button) Control button single press (1) or multipress (0)
    2 (OFF) (MQTT) Add global temperature/humidity/pressure info to JSON sensor message
    3 (ON ) (MQTT) Enable (1)
    4 (OFF) (MQTT) Switch between RESULT (0) or COMMAND (1)
    5 (OFF) (MQTT) Retain on Power
    6 (OFF) (MQTT) Retain on Button
    7 (OFF) (MQTT) Retain on Switch
    8 (OFF) (Temperature) Switch between Celsius (0) or Fahrenheit (1)
    9 (OFF) (MQTT) Retain on Sensor
   10 (OFF) (MQTT) Control LWT message format, Offline (1) or none (0)
   11 (OFF) (Button) Swap (1) button single and double press functionality
   12 (OFF) (Settings) Switch between dynamic (0) or fixed (1) slot flash save location
   13 (OFF) (Button) Support only single press (1) to speed up button press recognition
   14 (OFF) (Interlock) Power interlock mode
   15 (ON ) (Light) Switch between commands PWM (0) or COLOR/DIMMER/CT/CHANNEL (1)
   16 (OFF) (WS2812) Switch between clockwise (0) or counter-clockwise (1)
   17 (OFF) (Light) Switch between decimal (1) or hexadecimal (0) output
   18 (OFF) (Light) Pair light signal (1) with CO2 sensor
   19 (OFF) (HAss) Control automatic discovery (1) (See SetOption59)
   20 (OFF) (Light) Control power in relation to Dimmer/Color/Ct changes (1)
   21 (OFF) (Energy) Show voltage even if powered off (1)
   22 (OFF) (MQTT) Serial
   23 (OFF) (MQTT) Serial binary
   24 (OFF) (Pressure) switch between hPa (0)or mmHg (1) unit
   25 (OFF) (KNX) Enabled
   26 (OFF) (MQTT, HAss) Switch between POWER (0) or POWER1 (1)
   27 (OFF) (KNX) Enhancement
   28 (OFF) (RF) Receive data format hexadecimal (0) or decimal (1)
   29 (OFF) (IR) Receive data format hexadecimal (0) or decimal (1)
   30 (OFF) (HAss) enforce autodiscovery as light (1)
   31 (OFF) (Wifi, MQTT) Control link led blinking (1)
   32 ( 40) (Button/Switch) Key hold time detection in decaseconds (default 40)
   33 (  5) (Energy) Maximum number of retries before deciding power limit overflow (default 5)
   34 (200) (Backlog) Minimal delay in milliseconds between executing backlog commands (default 200)
   35 (  0) (not used) mDNS delayed start (Sec)
   36 (  1) (Restart) Number of restarts to start detecting boot loop (default 1)
   37 (  0) (Light) RGB and White channel separation (default 0)
   38 (  6) (IR) Set the smallest sized UNKNOWN message packets we actually care about (default 6, max 255)
   39 (  0) (CSE7766) Number of invalid power measurements before declaring it invalid allowing low load measurments (default 128)
   40 (  0) (Button/Shutter) Ignore button change in seconds (default 0)
   41 ( 60) (Wifi) Interval in seconds between gratuitous ARP requests (default 60)
   42 ( 90) (Energy) Turn all power off at or above this temperature (default 90C)
   43 ( 10) (Rotary) Rotary step boundary (default 10)
   44 ( 25) (IR) Base tolerance percentage for matching incoming IR messages (default 25, max 100)
   45 ( 40) (Bistable) Pulse time in milliseconds for two coil bistable latching relays (default 40)
   46 (  0) (PowerOn) Add delay of 10 x value milliseconds at power on
   47 (  0) (PowerOn) Add delay of value seconds at power on before activating relays
   48 (  0) (not used) Energy Tariff2 start hour
   49 (  0)
   50 (OFF) (Timers) Enabled
   51 (OFF) (GPIO) Enable ESP8285 user GPIO's (1)
   52 (OFF) (Time) Append timezone to JSON time (1)
   53 (ON ) (GUI) Show hostname and IP address in GUI main menu (1)
   54 (OFF) (Tuya) Apply SetOption20 settings to Tuya device (1)
   55 (OFF) (mDNS) Service on (1) or off (0)
   56 (OFF) (Wifi) Scan network at restart for configured AP's (1) or used stored AP (0)
   57 (ON ) (Wifi) Scan network every 44 minutes for configured AP's (1)
   58 (OFF) (IR) Add IR Raw data to JSON message (1)
   59 (OFF) (MQTT) Send tele/%topic%/STATE in addition to stat/%topic%/RESULT (1)
   60 (OFF) (Sleep) Enable normal sleep (1) instead of dynamic sleep (0)
   61 (OFF) (Button, Switch) Force local operation (1) when button/switch topic is set
   62 (OFF) (MQTT) Don't use retain flag on HOLD messages (1)
   63 (OFF) (Power) Don't scan relay power state at restart (1)
   64 (OFF) (JSON) Enable '_' (1) instead of '-' (0) as sensor index separator
   65 (OFF) (QPC) Disable (1) fast power cycle detection for device reset
   66 (OFF) (Tuya) Enable (1) TuyaMcuReceived messages over Mqtt
   67 (OFF) (Buzzer) Enable (1) buzzer when available
   68 (OFF) (Light) Enable multi-channels PWM (1) instead of Color PWM (0)
   69 (OFF) (not used) Limits Tuya dimmers to minimum of 10% (25) when enabled
   70 (OFF) (Energy) Enable Weekend Energy Tariff
   71 (OFF) (DDS2382) Select different Modbus registers (1) for Active Energy (#6531)
   72 (OFF) (Energy) Enable (1) hardware energy total counter as reference (#6561)
   73 (OFF) (Button) Detach buttons from relays (1) and enable MQTT action state for multipress
   74 (OFF) (DS18x20) Enable internal pullup (1) for single DS18x20 sensor
   75 (OFF) (MQTT) GroupTopic replaces %topic% (0) or fixed topic cmnd/grouptopic (1)
   76 (OFF) (Deepsleep) Enable incrementing bootcount (1) when deepsleep is enabled
   77 (OFF) (Light) Do not power off (1) if slider moved to far left
   78 (OFF) (not used) Disable OTA compatibility check
   79 (OFF) (Counter) Enable resetting of counters (1) after telemetry was sent
   80 (OFF) (Shutter) Enable shutter support (1)
   81 (OFF) (PCF8574) Invert all ports on PCF8574 devices (1)
   82 (OFF) (Alexa) Reduced CT range for Alexa (1)
   83 (OFF) (Zigbee) Use FriendlyNames (1) instead of ShortAddresses (0) when possible
   84 (OFF) (AWS IoT) publish MQTT state to a device shadow (1)
   85 (OFF) (DevGroups) Enable Device Groups (1)
   86 (OFF) (PWM Dimmer) Turn brightness LED's off (1) 5 seconds after last change
   87 (OFF) (PWM Dimmer) Turn red LED on (1) when powered off
   88 (OFF) (DevGroups) Enable relays in separate device groups/PWM Dimmer Buttons control remote devices (1)
   89 (OFF) (MQTT, Zigbee) Distinct MQTT topics per device for Zigbee (1) (#7835)
   90 (OFF) (MQTT) Disable non-json MQTT response (1)
   91 (OFF) (Light) Enable light fading at start/power on (1)
   92 (OFF) (Light) Set PWM Mode from regular PWM (0) to ColorTemp (1) control (Xiaomi Philips ...)
   93 (OFF) (Compress) Keep uncompressed rules in memory to avoid CPU load of uncompressing at each tick (1)
   94 (OFF) (MAXxxx) Implement simpler MAX6675 protocol (1) instead of MAX31855 (0)
   95 (ON ) (Wifi) Enable Wifi
   96 (ON ) (Eth) Enable Ethernet (ESP32)
   97 (OFF) (Tuya) Set Baud rate for TuyaMCU serial communication to 9600 (0) or 115200 (1)
   98 (OFF) (Rotary) Use rules (1) instead of light control (0)
   99 (OFF) (PWM Dimmer) Enable zerocross dimmer (1)
  100 (OFF) (Zigbee) Remove ZbReceived form JSON message (1)
  101 (OFF) (Zigbee) Add the source endpoint as suffix to attributes, ex `Power3` (1) instead of `Power` (0) if sent from endpoint 3
  102 (OFF) (Teleinfo) Set Baud rate for Teleinfo communication to 1200 (0) or 9600 (1)
  103 (OFF) (MQTT TLS) Enable TLS mode (1) (requires TLS version)
  104 (OFF) (MQTT) No Retain (1) - disable all MQTT retained messages, some brokers don't support it: AWS IoT, Losant
  105 (OFF) (Light) White Blend Mode (1) - used to be `RGBWWTable` last value `0`, now deprecated in favor of this option
  106 (OFF) (Light) Virtual CT (1) - Creates a virtual White ColorTemp for RGBW lights
  107 (OFF) (Light) Virtual CT Channel (1) - signals whether the hardware white is cold CW (true) or warm WW (false)
  108 (OFF) (Teleinfo) Enable Teleinfo + Tasmota Energy device (0) or Teleinfo raw data only (1)
  109 (OFF) (Alexa) Gen1 mode (1) - if you only have Echo Dot 2nd gen devices
  110 (OFF) (Zigbee) Disable auto-config (1) when pairing new devices
  111 (OFF) (Buzzer) Use frequency output (1) for buzzer pin instead of on/off signal (0)
  112 (OFF) (Zigbee) Use friendly name in zigbee topic (1) (use with SetOption89)
  113 (OFF) (Rotary) Set dimmer low on rotary dial after power off (1)
  114 (ON ) (Switch) Detach Switches from relays and enable MQTT action state for all the SwitchModes (1)
  115 (OFF) (ESP32 BLE) Enable ESP32 MI32 BLE (1)
  116 (OFF) (Zigbee) Disable auto-query of zigbee lights and devices (1)
  117 (OFF) (Light) run fading at fixed duration instead of fixed slew rate
  118 (OFF) (Zigbee) Move ZbReceived from JSON message and into the subtopic replacing SENSOR default
  119 (OFF) (Zigbee) Remove the device addr from json payload, can be used with zb_topic_fname where the addr is already known from the topic
  120 (OFF) (Zigbee) Append endpoint number to topic if device dependent (use with SetOption89)
  121 (OFF) (MQTT) Retain on State
  122 (OFF) (MQTT) Retain on Info
  123 (OFF) (Wiegand) switch tag number output to hex format (1)
  124 (OFF) (Wiegand) send key pad stroke as single char (0) or one tag (ending char #) (1)
  125 (OFF) (Zigbee) Hide bridge topic from zigbee topic (use with SetOption89) (1)
  126 (OFF) (DS18x20) Enable arithmetic mean over teleperiod for JSON temperature (1)
  127 (OFF) (Wifi) Keep wifi in no-sleep mode, prevents some occasional unresponsiveness
  128 (ON ) (Web) Allow access without referer check
  129 (OFF) (Energy) Show phase information
  130 (OFF) (Debug) Show heap with logging timestamp
  131 (OFF) (Tuya) Allow save dimmer = 0 receved by MCU
  132 (OFF) (TLS) Use fingerprint validation instead of CA based
  133 (OFF) (Shift595) Invert outputs of 74x595 shift registers
  134 (OFF) (PWM) force PWM lights to start at same phase, default is to spread phases to minimze overlap (also needed for H-bridge)
  135 (OFF) (Display & LVGL) force disabling default splash screen
  136 (OFF) (TuyaSNS) When ON disable publish single SNS value on Tuya Receive (keep Teleperiod)
  137 (OFF) (Tuya) When Set, avoid the (mqtt-) publish of Tuya MCU Heartbeat response if SetOption66 is active
  138 (OFF) (GUI) Align (energy) table values left (0) or right (1)
  139 (OFF) (Pressure) Switch between mmHg (0) or inHg (1) when SO24 1
  140 (OFF) (MQTT) MQTT clean session (0 = default) or persistent session (1)
  141 (OFF) (GUI) Disable display of GUI module name (1)
  142 (OFF) (Wifi) Wait 1 second for wifi connection solving some FRITZ!Box modem issues (1)
  143 (OFF) (Zigbee) Disable Battery auto-probe and using auto-binding
  144 (OFF) (Zigbee) Include time in `ZbReceived` messages like other sensors
  145 (OFF) (MQTT) Retain on Status
  146 (OFF) (ESP32) Show ESP32 internal temperature sensor
  147 (OFF) (MQTT) Disable publish SSerialReceived MQTT messages, you must use event trigger rules instead
  148 (OFF) (Light) start DMX ArtNet at boot, listen to UDP port as soon as network is up
  149 (OFF) (Wifi) prefer IPv6 DNS resolution to IPv4 address when available. Requires `#define USE_IPV6`
  150 (OFF) (Energy) Force no voltage/frequency common
  151 (OFF) (Matter) Enable Matter protocol over Wifi
  152 (OFF) (Power) Switch between two (0) or one (1) pin bistable relay control
  153 (OFF) (Berry) Disable autoexec.be on restart (1)

Features
  Language LCID = 1046
  MQTT_PUBSUBCLIENT
  USE_ADC_VCC
  USE_ARILUX_RF
  USE_ARMTRONIX_DIMMERS
  USE_BP1658CJ
  USE_BP5758D
  USE_BUZZER
  USE_CONFIG_OVERRIDE
  USE_COUNTER
  USE_DEEPSLEEP
  USE_DEVICE_GROUPS
  USE_EXS_DIMMER
  USE_IMPROV
  USE_LIGHT
  USE_MODBUS_BRIDGE
  USE_PING
  USE_PWM_DIMMER
  USE_SCRIPT
  USE_SERIAL_BRIDGE
  USE_SHELLY_DIMMER
  USE_SM16716
  USE_SM2135
  USE_SM2335
  USE_SONOFF_D1
  USE_SONOFF_IFAN
  USE_SONOFF_L1
  USE_SONOFF_RF
  USE_SPI
  USE_UFILESYS
  USE_WEBSERVER
  VTABLES_IN_FLASH
  WEBSERVER_ADVERTISE

**Comment by sfromis** [comment]
Well, there are quite a few features enabled which you likely are not using. Still, most drivers/features you can elide are not using a lot of RAM until actually being active.

**Reply by DC-555** [reply]
I narrowed it down further, and now it works. But I believe what resolved it was something else. I modified the way I activated it in user_config_override.h I did it like this:
#ifdef USE_SENDMAIL
#undef USE_SENDMAIL
#endif
#define USE_SENDMAIL

before it was just like this:
#define USE_SENDMAIL
#define USE_TLS

Features
  Language LCID = 1046
  MQTT_PUBSUBCLIENT
  USE_ADC_VCC
  USE_CONFIG_OVERRIDE
  USE_COUNTER
  USE_LIGHT
  USE_MODBUS_BRIDGE
  USE_PING
  USE_RDM6300
  USE_SCRIPT
  USE_SPI
  USE_UFILESYS
  USE_WEBSERVER

**Reply by barbudor** [reply]
That change in activation have no impact on code/ram usage
A module is either built-in or not

Despite what sfromis says, although I haven't look at the code for each in details, old designed modules/drivers tends to allocate ram as global variable even when not in use.

**Comment by DC-555** [comment]
Yes, I thought I applied, but not yet. while I don't put the script it works, but after I add the script it doesn't work anymore. Even if I delete or disable.
